crypto/cmp/cmp_ctx.c

```diff
@@ -120,6 +125,28 @@ OSSL_CMP_CTX *OSSL_CMP_CTX_new(void)
     return NULL;
 }
 
+/*
+ * prepare the OSSL_CMP_CTX for next use, partly re-initializing OSSL_CMP_CTX
+ */
+int OSSL_CMP_CTX_reinit(OSSL_CMP_CTX *ctx)
+{
+    if (ctx == NULL)
+        return 0;
+
+    ctx->log_level = -1;
+    ctx->lastPKIStatus = -1;
+    ctx->failInfoCode = -1;
+
+    return CMP_CTX_set0_statusString(ctx, NULL)
```

Comment:
Why are there two different prefixes for the "methods" of "class" `OSSL_CMP_CTX`?

For some, it's `OSSL_CMP_CTX_`, for others it's `CMP_CTX_`.

Reply:
The public functions (exported via cmp.h and cmp_util.h) are required to have a `OSSL_` prefix by OpenSSL guidelines,
and all functions related to the CMP context have `CMP_CTX_` in their name.

The CMP-internal context-related functions usually just have the `CMP_CTX` prefix (while there are some of them that still have the `OSSL_` prefix for historic reasons since they were public but then we decided to make them internal in order to simplify the CMP API).

Reply:
Oh, I understand. Nevertheless, I'm a bit torn apart about that question.

- On one hand: If the functions are not public, than they should be in lowercase for consistency reasons. It does not matter that they used to be uppercase. In particular, since they never were public in OpenSSL, only in your fork. OpenSSL already has far too many inconsistencies for 'historic' reasons. And believe me, I know what it means to 'unpublish' an entire class interface, watch out for 'RAND_POOL' in  https://github.com/openssl/openssl/commit/6decf9436f77ff65ed8ed773268663a9273cfbc8#diff-baca3d0b945f277b98ef9b654387b6bf.(*)
- On the other hand: I understand that you probably have a large code base which relies on the capitalized names and that we might be asking too much of you to change all your code. After all, we are grateful for your contribution.

This is a question that I am not able to decide. @mattcaswell @bernd-edlinger, what is your opinion?

_(*) Conversion to lowercase was actually a straightforward task using find and sed._

Reply:
Thanks for your understanding response.

Changing the names of internal functions usually is not too hard since the documentation and the nasty `util/libcypto.num` is not affected, though the meta files like `cmperr.h` usually auto-generated by `make errors` typically need manual treatment.
So we can get rid of needless `OSSL_` prefixes for them without very much effort.

What I really do not like to do is to convert `CMP_CTX_` prefixes to lower-case `cmp_ctx_` (and similarly many more such as `OSSL_CMP_MSG_`) for a different reason: consistency with the name of the type `CMP_CTX` (as well as types `OSSL_CMP_MSG` etc. defined mostly in `crypto/cmp/cmp_asn.c`)

Reply:
Note: I changed my mind and think now that uppercase is ok here. It's a case of conflicting naming conventions, see my https://github.com/openssl/openssl/pull/9127#discussion_r304765574.

Reply:
Fixed